src/algorithm/nest_cfgs.rs

```diff
@@ -428,8 +428,8 @@ pub(crate) mod test {
         //               \-> right -/             \-<--<-/
         let mut cfg_builder = CFGBuilder::new(FunctionType::new(type_row![NAT], type_row![NAT]))?;
 
-        let pred_const = cfg_builder.add_constant(Const::simple_predicate(0, 2))?; // Nothing here cares which
-        let const_unit = cfg_builder.add_constant(Const::simple_unary_predicate())?;
+        let pred_const = cfg_builder.add_constant(Const::simple_predicate(0, 2), None)?; // Nothing here cares which
```

Comment:
Good to update the comment here - "Nothing here cares which *branch*" to be clear that the comment isn't about the extensions. (However, I think this should not actually be showing up in this PR, so this might disappear post-merge)